### PR TITLE
Upgrade to new mocks (resolve #155, #156)

### LINKED ISSUE #155: Left sidebar
The sidebar is moved to the left side with new styles and icons.

image

Invision mock:
https://egghead.invisionapp.com/d/main#/console/10461992/222542106/preview

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Navigation on left with new mobile slide out and avatar
- Screens broken up into `Widget` and `List` items
- Change `Overview` to `Dashboard` with its own screen
- Collapse `New` (request/proposing) into `Dashboard`; it will come back when we have full lesson uploading
- Update `Instructor` screen to make more sense
- Add new `LayoutSplit` component that stacks vertically on mobile and horizontally as a grid on desktop
- Remove restrictive `Screen` component in favor of widgets and `Layout*` components
- Made some updates to `tachyons-egghead` and `egghead-ui` along the way

To review this, it might be easiest to just check it out or look through the directory structure here:
https://github.com/eggheadio/egghead-instructor-center/tree/widgets/src

# Result For User

Doesn't match [the mocks](https://egghead.invisionapp.com/d/main#/console/10461992/222244883/preview) fully yet, but is headed in the right direction I think. I like it.

widgets

---

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +16,11 @@ deploy:
...
 notifications:
-  slack: $SLACK_INTEGRATION_TOKEN
+  email:
+    on_success: never
+    on_failure: always
+  slack:
+    on_success: never
+    on_failure: always
+    rooms:
+      secure: LnEL+23Jv8ts6tufKfvKoRT81c8xvEOd+ZvErqmgXPIb5HG525eudD8WaOsAF1XKxSq74JRRYp9y5jSC6/r3TDBDsRNV9sW4U/0/peLnSF/yzBII+W2JEUFe9RdLNrVSMxQRfvBesl3QlHRxOJ1Jhz4QYaPAodsTqDPUrUiW6QA=
```
Comment (trevordmiller) [inline .travis.yml:26]: This should fix travis and get it to only send emails and slack notifications in the #egghead-travis channel on failure. I did the same thing in `egghead-ui` and `tachyons-egghead`.

File: README.md [hunk 2]
```diff
@@ -34,21 +34,21 @@ Each directory inside `src` is a **component**. A component is a directory organ
...
-    utils/ (optional utility modules)
+  index.js (entry point)
+  components/ (optional sub-components)
+  utils/ (optional utility modules)
 ```
 
 ## Screens
 
-Some components are also **screens**. A screen is just a component that additionally corresponds with a route and is generally connected to endpoints. They are also known as "Smart", "Container", or "Connected" components. They look like this:
+Some components are also **screens**. A screen is just a component that additionally corresponds with a route; it is generally a collection of `Widget`s. They look like this:
```
Comment (trevordmiller) [inline README.md:44]: Screens are now generally collections of "Widgets" which are just a component with the card style box around it, like the new mocks. A few new components have been added to this PR for this (`Widget`, `List`, `Layout*` and `*Widget` pieces).

File: public/index.html [hunk 3]
```diff
@@ -19,7 +19,7 @@
       ga('send', 'pageview');
     </script>
   </head>
-  <body>
+  <body class='bg-tag-gray'>
```
Comment (trevordmiller) [inline public/index.html:22]: Light gray background to override `tachyons-egghead` `bg-navy`

File: src/components/InstructorStatsWidget/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <Widget title={statsTitleText}>
+      <List items={map(items, (item, index) => (
+        <IconLabel
+          iconType={item.type}
+          labelText={item.text}
+        />
+      ))} />
+    </Widget>
```
Comment (trevordmiller) [inline src/components/InstructorStatsWidget/index.js:33]: Most `screen/*/components` files are widgets like this now; they are self-contained with all the sub-components, utilities and `WrappedRequests` when needed so they can be moved or promoted simply by moving the directory.

File: src/components/LayoutColumns/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+export default ({items}) => (
+  <div className='flex-ns'>
+    {map(items, (item, index) => (
+      <div
+        key={index}
+        className={`w-100-ns ${index < items.length - 1 ? 'mr4-ns' : ''}`}
+      >
+        {item}
+      </div>
+    ))}
```
Comment (trevordmiller) [inline src/components/LayoutColumns/index.js:13]: This component lets you pass a list of `items` and it makes them into a horizontal grid on larger screen sizes with margin "gutter" between, except on the last item.

File: src/components/LessonState/index.js [hunk 6]
```diff
@@ -11,9 +12,9 @@ export default ({lesson}) => {
       `}>
         {lesson.state}
       </div>
-      <div className="white-50">
+      <Paragraph>
         {detailsByLessonState[lesson.state].description}
-      </div>
+      </Paragraph>
```
Comment (trevordmiller) [inline src/components/LessonState/index.js:17]: Using the latest typography elements from `egghead-ui`

File: src/components/List/index.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    {map(items, (item, index) => (
+      <div
+        key={index}
+        className={`pa4 ${index < items.length - 1 ? 'bb b--light-gray' : ''}`}
+      >
+        {item}
+      </div>
+    ))}
+  </div>
+)
```
Comment (trevordmiller) [inline src/components/List/index.js:15]: This adds "cells" of items in a list; generally used inside a widget so it adds padding and a bottom border to all items but the last:

image

File: src/components/InstructorStatsWidget/index.js (hunk 4 above)
Comment (trevordmiller) [reply src/components/InstructorStatsWidget/index.js:33]: image

Comment (trevordmiller) [reply src/components/InstructorStatsWidget/index.js:33]: They all compose like this: 

image

File: src/components/LayoutColumns/index.js (hunk 5 above)
Comment (trevordmiller) [reply src/components/LayoutColumns/index.js:13]: image

File: src/components/LessonState/index.js (hunk 6 above)
Comment (trevordmiller) [reply src/components/LessonState/index.js:17]: image

File: src/components/Main/index.js [hunk 8]
```diff
@@ -1,8 +1,10 @@
 import React, {PropTypes} from 'react'
 
 const Main = ({children}) => (
-  <main className='pv5 ph4 mw8 center'>
-    {children}
+  <main className='bg-tag-gray h-100 w-100 dark-gray ml7-ns mt4 mt0-ns'>
+    <div className='mw8 pv4 ph3 ph4-ns'>
+      {children}
+    </div>
```
Comment (trevordmiller) [inline src/components/Main/index.js:7]: New styles for the light background and navigation on the left

File: src/components/Navigation/index.js [hunk 9]
```diff
@@ -45,86 +74,109 @@ export default class Navigation extends Component {
...
 
-        </div>
+          })}
+        </nav>
 
-      </header>
+      </aside>
     )
   }
 }
```
Comment (trevordmiller) [inline src/components/Navigation/index.js:182]: New styles and logic for the slide out left menu

File: src/components/Screen/index.js [hunk 10]
```diff
@@ -1,43 +0,0 @@
...
-)
-
-Split.propTypes = {
-  intro: PropTypes.node,
-  title: PropTypes.string.isRequired,
-  main: PropTypes.node.isRequired,
-  aside: PropTypes.node,
-}
-
-export default Split
```
Comment (trevordmiller) [inline src/components/Screen/index.js:43]: Not using this component anymore; now screens `*/screen/*` are mostly composed of `Widget`, and `Layout*` components.

File: src/components/Widget/index.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+      ? <Paragraph type='small'>
+          {description}
+        </Paragraph>
+      : null
+    }
+    <div className='bg-white shadow-1 br2'>
+      {children}
+    </div>
+  </section>
+)
```
Comment (trevordmiller) [inline src/components/Widget/index.js:19]: Used a lot now. Gives a title, optional description, and renders children; children are often wrapped in `List`:
image

Comment (trevordmiller) [reply src/components/Widget/index.js:19]: Could be a good candidate for promotion to `egghead-ui`

File: src/components/List/index.js (hunk 7 above)
Comment (trevordmiller) [reply src/components/List/index.js:15]: Could be a good candidate for promotion to `egghead-ui`

File: src/components/LayoutColumns/index.js (hunk 5 above)
Comment (tayiorbeii) [reply src/components/LayoutColumns/index.js:13]: nice!

**Review by tayiorbeii — APPROVED** [review]

**Review by joelhooks — COMMENTED** [review]
First off, this is awesome, love that we took some fresh designs and applied them like this. 👍

> with the card style box 

These are generally an evolution of the `Card`. We are using `Card` and layouts. The cards we've been using are specifically content oriented, but as described they seem conceptually very similar. 

I'm fully [bike-shedding](https://en.wikipedia.org/wiki/Law_of_triviality) here, because it doesn't really matter but a couple of things stand out.

The word `Widget` deserves some consideration, as we are going to be stuck with it and I'd like to keep these common definitions, well, common. If these are "card style boxes", perhaps `Card` is more specific and descriptive in the context of our development process? It's what we've been using, and the way designers refer to them.

In this refresh, I see a `TitleCard` base component used for modular elements. I like this descriptive name over the very generic `Widget`.

My other concern is with the naming of the individual cards. `InstructorStatsWidget` could just be `InstructorStats` and we don't lose any important information. I'm generally a fan of not notating the type in file names, unless it delivers value in describing the item.

All in all, these are trivial concerns, and I'm just nitpicking. This is a great step in the right direction. If widgets is strongly preferred I will learn to embrace it over time 😄

**Comment by trevordmiller** [comment]
@joelhooks Thanks. I like `Card` better than `Widget`; I just used widget because it was the terminology I was seeing in slack and the mocks, but I'll happily change it. So @vojtaholik, we'll use "Card" now instead of "Widget" to refer to the blocks on each "screen". As for naming the individual cards with `*Widget/Card` at the end, I went back and forth on this, but decided it would be good to include it because it conveys that it is a single component that can't really be composed in anything other than a screen. For example, I have a `LessonListsByStates` which can be used in any other component, but when wrapped with a title and specific props, it becomes `InstructorLessonsCard` (https://github.com/eggheadio/egghead-instructor-center/blob/master/src/components/InstructorLessonsWidget/index.js) to convey it has the card styles; there could be an `InstructorLessons` component that is the same thing without the Card wrapper. If I did drop the `*Card` from the end, I would move the `<Card>` wrappers from the components themselves and put them in the screens, which is probably a better approach anyways and solves the naming problem. So instead of a screen being something like:

```
  <div>
      <SomeThing1Card />
      <SomeThing2Card />
  </div>
```

It could be:

```
  <div>
      <Card title={someTitle}>
        <SomeThing1 />
      </Card>
      <Card title={someTitle}>
        <SomeThing2 />
      </Card>
  </div>
```

Added a refactor story for this: https://github.com/orgs/eggheadio/projects/7

**Pull request merged**